### PR TITLE
feat: update node version

### PR DESCRIPTION
**Pull request by harish-chandra-89**
Updated node 8 to 12

### CODE REVIEW

**Comment by harish-chandra-89** [comment]
Build pipeline is failed. It seems like it's target only master branch. Please have a look and suggest if I'm wrong here.

**Comment by mattcan** [comment]
Please fix the pipeline. It looks like the previous way of getting NPM working in Github actions has been deprecated. Documentation to fix is here: https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-nodejs-with-github-actions#installing-dependencies

**Comment by harish-chandra-89** [comment]
@mattcan I have tried as per your suggestion but its not resolved. Please check the latest commit.

**Comment by mattcan** [comment]
> Tried to get the reference from other repo but I've doubt in "uses: actions/npm@master".

Yes, the documentation seems to be a replacement for any `uses: actions/npm@master` sections. Case in point, you have added a step for `npm i` but right after is another step that does the same except using an action that doesn't exist.

**Comment by harish-chandra-89** [comment]
@mattcan Build pipeline fixed

File: .github/workflows/build-publish.yml [hunk 1]
```diff
@@ -11,33 +11,38 @@ jobs:
...
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
         with:
-          args: i
+          node-version: '12.x'
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install dependencies
+        run: npm i
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
```
Comment (mattcan) [inline .github/workflows/build-publish.yml]: `NPM_AUTH_TOKEN` doesn't seem to be a secret that we have. A _quick_ look over the dependencies suggests that there is no need for a secret. Can you confirm?

Comment (harish-chandra-89) [reply .github/workflows/build-publish.yml]: @mattcan Should I need to make changes as suggested in https://github.com/telus/express-locale/pull/36

Comment (maumercado) [reply .github/workflows/build-publish.yml:22]: @harish-chandra-89 NODE_AUTH_TOKEN does not exists

File: .github/workflows/build-publish.yml [hunk 2]
```diff
@@ -11,33 +11,38 @@ jobs:
         name: checkout code
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: npm i
-        uses: actions/npm@master
+      - uses: actions/checkout@v1
```
Comment (maumercado) [inline .github/workflows/build-publish.yml:14]:
```suggestion
      - uses: actions/checkout@v2
```

Comment (harish-chandra-89) [reply .github/workflows/build-publish.yml]: @maumercado Suggested changes are updated, Please have a look

**Pull request merged**